### PR TITLE
GH-2792: HEnrichSpec: Add hooks for common headers

### LINKED ISSUE #2792: Add convenient options into the HeaderEnricherSpec for common headers
At least reflect what `<int:header-enricher>` provides:

* correlation-id
* error-channel
* reply-channel
* expiration-date
* priority
* routing-slip

as well as we may consider to add hook for the `IntegrationMessageHeaderAccessor.SEQUENCE_NUMBER` and `IntegrationMessageHeaderAccessor.SEQUENCE_SIZE`.

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-integration/issues/2792

### CODE REVIEW

**Comment by artembilan** [comment]
If this is OK, I'll add other common header hooks.

It's just too much copy/paste and extra code to support in the future...

**Comment by artembilan** [comment]
Ready for review.

Thanks

File: spring-integration-core/src/main/java/org/springframework/integration/dsl/HeaderEnricherSpec.java [hunk 1]
```diff
@@ -528,6 +529,206 @@ public HeaderEnricherSpec errorChannelExpression(String errorChannelExpression,
...
+	 * If the header exists, it will <b>not</b> be overwritten
+	 * unless {@link #defaultOverwrite(boolean)} is true.
+	 * @param expirationDateFunction the function.
+	 * @param <P> the payload type.
+	 * @return the header enricher spec.
+	 * @see FunctionExpression
+	 * @since 5.2
+	 */
+	public <P> HeaderEnricherSpec expirationDateFunction(Function<Message<P>, Object> expirationDateFunction) {
+		return priorityFunction(expirationDateFunction, null);
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/dsl/HeaderEnricherSpec.java]: copy/paste

**Review by garyrussell — CHANGES_REQUESTED** [review]
I feel your pain 😄

**Review by garyrussell — APPROVED** [review]

**Pull request merged**